tools/collect-readme.py

```diff
@@ -0,0 +1,169 @@
+            pass
+        else:
+            #print(file)
+            processed.append(file)
+            with open(results[0], 'r', encoding='utf-8') as f:
+                md_content = f.read()
+
+            # Parse front matter and content
+            path_obj = Path(results[0])
+            slug_obj = Path(file)
+            folder4 = list(slug_obj.parts)[4]
+            slug = '/readmes/'+folder4+'/'+list(slug_obj.parts)[5] #Gets the slug for the MDX file, based on the file path
+            if folder4 == 'module':
+                folder4 = ''
+            else:
+                folder4 = folder4+' '
+            markdown_content = md_content_first+parent_folder.title()+md_content_second+list(slug_obj.parts)[-3].title()+md_content_third+list(slug_obj.parts)[-3]+' '+folder4+md_content_fourth+slug+md_content_fifth
+            file_path = Path(file)
+            item_path2 = list((Path(file)).parts)
+            #print(list(file_path.parts))
+
+            name_no_md = list(file_path.parts)[5].removesuffix('.md')
+            with open(file,'r') as firstfile, open('/home/nubots/NUbook/src/book/04-readmes/'+list(slug_obj.parts)[4]+'/'+name_no_md+'.mdx','a+') as secondfile:
+
+                try: #To determine if header is already in there, if not add in header
+                    file_search = mmap.mmap(secondfile.fileno(), 0, access=mmap.ACCESS_READ)
+                    if file_search.find(b'---') != -1:
+                        pass
+                    else:
+                        secondfile.write(markdown_content)
+                except ValueError:
+                    secondfile.write(markdown_content)
+
+                #Takes out the corrupted banner refrence from the Nuclear readme.
+                #If that specific line is found it will be replaced, if not it is written/copied over normally.
+                try:
+                    first_file_search = mmap.mmap(firstfile.fileno(), 0, access=mmap.ACCESS_READ)
+                    filelines = firstfile.readlines()
+                    counter = 0
+                    for each_line in filelines:
+                        counter += 1
+                        if each_line.find('your own banner') != -1:
+                            print(each_line)
+                            secondfile.write('If you do not set your own banner file, !the default banner will be used')
+                        else:
+                            secondfile.write(each_line)
+
+                except:
+                    print('nah')
+                '''
+                # read content from first file
+                for lines in firstfile:
+                    # append content to second file
+                    secondfile.write(lines)'''
+                firstfile.close()
+                secondfile.close()
+    return
+
+list_of_dirs = ['/home/nubots/NUbots/module', '/home/nubots/NUbots/nuclear', '/home/nubots/NUbots/nusight2', '/home/nubots/NUbots/tools', '/home/nubots/NUbots/roles']
```

Comment:
Make these paths relative the script location

Reply:
I'd say take all paths as arguments the script: path to NUbots root (from which you can derive the sub paths to modules, NUsight, etc) and path to NUbook for output.

Long term though I think this is something that should live in NUbook, which can be triggered by a merge to NUbots main. That will then pull the latest NUbots changes and build it into NUbook, without committing any of the intermediate markdown files. All done in NUbook CI, with the only external input being a merge to NUbots main that triggers the NUbook build.